analyses/hotspots-detection/01-create-hotspot-maf.Rmd

```diff
@@ -0,0 +1,246 @@
+                         "results")
+# Make output folder
+if (!dir.exists(results_dir )) {
+  dir.create(results_dir , recursive = TRUE)
+}
+
+scratch_dir <- file.path(
+  root_dir,
+  "scratch",
+  "hotspot-detection"
+)
+
+# Import function to split MNV to SNV
+source(file.path(root_dir, "analyses", "snv-callers", "util", "split_mnv.R"))
+
+maf_coltypes <- readRDS(file.path("input", "maf_coltypes.RDS"))
+# maf_coltypes is a named vector with column types information for each column in a maf file
+# More description about maf format is available
+# [here](https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/)
+
+# combined hotspot calls
+combined_hotspot_calls <- list.files(path = scratch_dir,pattern = ".RDS",full.names = TRUE) %>%
+  map_dfr(readRDS) 
+
+# MNVs are split into SNVs
+combined_hotspot_calls_mnv_split <- split_mnv(combined_hotspot_calls) %>%
+  # removing mvv_id and mnv_pos column
+  # since they uniquely numbered per caller variant calls
+  # in the merged combined_hotspot_calls
+  select(-mnv_id, -mnv_pos)
+
+combined_hotspot_calls <- combined_hotspot_calls %>%
+  # removing MNVs here
+  filter(!Variant_Type %in% c("DNP", "TNP", "ONP")) %>%
+  # adding the MNV split SNV alleles here 
+  union_all(combined_hotspot_calls_mnv_split)
+
+# hotspot mutation maf file output
+output_file <- file.path(results_dir,"pbta-snv-scavenged-hotspots.maf.tsv.gz")
+
+
+# PBTA consensus maf 
+consensus_maf_file <- read_tsv(file.path(data_dir,"pbta-snv-consensus-mutation.maf.tsv.gz"),
+                                col_types = maf_coltypes)
+
+```
+
+## Gather consensus maf file for hotspots
+
+The main tasks are:
+ - filter `combined_hotspot_calls` to calls that are not found in consensus calls
+ - gather read supports for REF and ALT from multiple callers adding values from any 1 caller in the order strelka2 > mutect2 > lancet > vardict
+ 
+### Filter hotspots not found in consensus calls
+```{r warning=FALSE }
+# These columns provide unique values 
+# to check for exact in the 2 dataframes
+join_cols <- c("Hugo_Symbol",
+               "Entrez_Gene_Id",
+               "NCBI_Build",
+               "Chromosome",
+               "Start_Position",
+               "End_Position",
+               "Strand",
+               "Variant_Classification",
+               "Variant_Type",
+               "Reference_Allele",
+               "Tumor_Seq_Allele1",
+               "Tumor_Seq_Allele2",
+               "Tumor_Sample_Barcode")
+
+# Filter combined_hotspot_calls to calls that are only
+# found from the hotspot scavenging analysis
+combined_maf_hotspots_filt <- combined_hotspot_calls %>%
+  anti_join(consensus_maf_file,
+          by = join_cols) %>%
+  unique()
+
+# gather and add caller info
+caller_counts <- combined_maf_hotspots_filt %>%
+  select(join_cols,caller) %>%
+  unique() %>% 
+  group_by_at(join_cols) %>%
+  summarise(caller_count = n(),
+            caller = toString(unique(caller))
+  ) %>%
+  ungroup()
```

Comment:
caller_count is the number of callers (strelka2,mutect2,vardict,lancet) that called the variant.

Reply:
Okay, I think I was confused by the placement of this chunk, since the `caller_counts` dataframe isn't used until much later. 

I was originally going to make a suggestion to not use `toString()` here, because it makes string separated on ", " that is a bit harder to parse to individual entries if that becomes needed. I'd lean toward a bare comma with no space or a semicolon:

```
caller = paste(unique(caller), collapse=';')
```

Reply:
ok I'll add that and move it closer to the plotting chunks

Reply:
It actually needs to be here since in the next steps the `caller` columns is uniqued by prefix of caller names "strelka2","mutect2","vardict" and "lancet" . I did look into `unite` https://tidyr.tidyverse.org/reference/unite.html to paste the `<caller_name>_caller`  columns using  but that adds NA for columns so that would not be ideal